src/misc/lv_array.c

```diff
@@ -45,71 +45,143 @@ void lv_array_init(lv_array_t * array, uint32_t capacity, uint32_t element_size)
 {
     array->size = 0;
 }
 
-void lv_array_resize(lv_array_t * array, uint32_t new_capacity)
+/**
+ * Remove the element at the specified position in the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @param index the index of the element to remove
+ */
+bool lv_array_pop(lv_array_t * array, uint32_t index)
+{
+    if(index >= array->size) {
+        return false;
+    }
+
+    /*Shortcut*/
+    if(index == array->size - 1) {
+        lv_array_resize(array, array->size - 1);
+        return false;
+    }
+
+    uint8_t * start = lv_array_at(array, index);
+    uint8_t * remaining = start + array->element_size;
+    uint32_t remaining_size = (array->size - index - 1) * array->element_size;
+    lv_memcpy(start, remaining, remaining_size);
+    lv_array_resize(array, array->size - 1);
+    return true;
+}
+
+bool lv_array_erase(lv_array_t * array, uint32_t start, uint32_t end)
 {
-    if(new_capacity > array->size) {
-        if(new_capacity > array->capacity) {
-            uint8_t * data = lv_malloc(new_capacity * array->element_size);
-            lv_memcpy(data, array->data, array->size * array->element_size);
-            lv_free(array->data);
-            array->data = data;
-            array->capacity = new_capacity;
-        }
+    if(end > array->size) {
+        end = array->size;
+    }
+
+    if(start >= end) {
+        return false;
+    }
+
+    /*Shortcut*/
+    if(end == array->size) {
+        lv_array_resize(array, start);
+        return true;
     }
-    else {
+
+    uint8_t * start_p = lv_array_at(array, start);
+    uint8_t * remaining = start_p + (end - start) * array->element_size;
+    uint32_t remaining_size = (array->size - end) * array->element_size;
+    lv_memcpy(start_p, remaining, remaining_size);
+    lv_array_resize(array, array->size - (end - start));
+    return true;
+}
+
+void lv_array_resize(lv_array_t * array, uint32_t new_capacity)
+{
+    uint8_t * data = lv_realloc(array->data, new_capacity * array->element_size);
+    LV_ASSERT_NULL(data);
+    array->data = data;
+    array->capacity = new_capacity;
+    if(array->size > new_capacity) {
         array->size = new_capacity;
     }
 }
 
-bool lv_array_append(lv_array_t * array, const uint8_t * element)
+bool lv_array_concat(lv_array_t * array, const lv_array_t * other)
 {
-    if(array->size >= array->capacity) {
-        // array is full
+    LV_ASSERT_NULL(array->data);
+    uint32_t size = other->size;
+    if(array->size + size > array->capacity) {
+        /*array is full*/
         return false;
     }
 
+    uint8_t * data = array->data + array->size * array->element_size;
+    lv_memcpy(data, other->data, array->element_size * size);
+    array->size += size;
+    return true;
+}
+
+bool lv_array_push_back(lv_array_t * array, const void * element)
+{
     LV_ASSERT_NULL(array->data);
-    uint8_t * data = (uint8_t *)(array->data + array->size * array->element_size);
+
+    if(array->size == array->capacity) {
```

Comment:
I think this push back function should increase automatically the array if capacity is reached, just like it is mentioned in STL vector function: 
https://cplusplus.com/reference/vector/vector/push_back/

So, then in other [PR](https://github.com/lvgl/lvgl/pull/5301/files#diff-52ff566468d87f6016ee458a4a89dffd561e014d31b1d16af51fd752570dd3afR99) you don't need to increase array capacity before calling push_back function.

Reply:
I was thinking the same, but in reality where array is used by lv_vector path, it normally doubles array size instead of increase one by one. So in the end, all APIs won't increase the array capacity automatically.

Reply:
I think both are valid use cases:
1. Having more control on how to update the capacity
2. Having a simple way to just append a value

 Can we have an other function like `lv_array_push_back_with_resize` ?

Reply:
I have changed `lv_array_push_back` to automatically resize with clear comment to state it.
For situations need better performance, a pre-resized array could be used.
For situations like lv_event, it can automatically resize.

Reply:
Pre-resize is a good idea!